modules/core/04-channel/keeper/multihop_timeout_test.go

```diff
@@ -97,67 +248,238 @@ func (suite *MultihopTestSuite) TestTimeoutPacket() {
+			chanCap = suite.A().Chain.GetChannelCapability(suite.A().ChannelConfig.PortID, suite.A().ChannelID)
+		}, false},
+		{"packet dest channel ID ≠ channel counterparty channel ID", false, func() {
+			suite.SetupChannels()
+			// use wrong channel for dest
+			*packet = types.NewPacket(ibctesting.MockPacketData, 1, suite.A().ChannelConfig.PortID, suite.A().ChannelID, suite.Z().ChannelConfig.PortID, ibctesting.InvalidID, defaultTimeoutHeight, disabledTimeoutTimestamp)
+			chanCap = suite.A().Chain.GetChannelCapability(suite.A().ChannelConfig.PortID, suite.A().ChannelID)
+		}, false},
+		{"connection not found - scenario 1", false, func() {
+			connectionIdx := 0
+			suite.A().SetupAllButTheSpecifiedConnection(uint(connectionIdx))
+			// pass channel check
+			suite.A().Chain.App.GetIBCKeeper().ChannelKeeper.SetChannel(
+				suite.A().Chain.GetContext(),
+				suite.A().ChannelConfig.PortID, suite.A().ChannelID,
+				types.NewChannel(types.OPEN, types.ORDERED, types.NewCounterparty(suite.Z().ChannelConfig.PortID, suite.Z().ChannelID), []string{connIDA}, suite.A().ChannelConfig.Version),
+			)
+			*packet = types.NewPacket(ibctesting.MockPacketData, 1, suite.A().ChannelConfig.PortID, suite.A().ChannelID, suite.Z().ChannelConfig.PortID, suite.Z().ChannelID, defaultTimeoutHeight, disabledTimeoutTimestamp)
+
+			// create chancap
+			suite.A().Chain.CreateChannelCapability(suite.A().Chain.GetSimApp().ScopedIBCMockKeeper, suite.A().ChannelConfig.PortID, suite.A().ChannelID)
+			chanCap = suite.A().Chain.GetChannelCapability(suite.A().ChannelConfig.PortID, suite.A().ChannelID)
+			packetHeight = suite.Z().Chain.LastHeader.GetHeight()
+			doUpdateClient = false
+		}, false},
+		{"connection not found - scenario 2", false, func() {
+			connectionIdx := 1
+			suite.A().SetupAllButTheSpecifiedConnection(uint(connectionIdx))
+			// pass channel check
+			suite.A().Chain.App.GetIBCKeeper().ChannelKeeper.SetChannel(
+				suite.A().Chain.GetContext(),
+				suite.A().ChannelConfig.PortID, suite.A().ChannelID,
+				types.NewChannel(types.OPEN, types.ORDERED, types.NewCounterparty(suite.Z().ChannelConfig.PortID, suite.Z().ChannelID), []string{connIDA}, suite.A().ChannelConfig.Version),
+			)
+			*packet = types.NewPacket(ibctesting.MockPacketData, 1, suite.A().ChannelConfig.PortID, suite.A().ChannelID, suite.Z().ChannelConfig.PortID, suite.Z().ChannelID, defaultTimeoutHeight, disabledTimeoutTimestamp)
+
+			// create chancap
+			suite.A().Chain.CreateChannelCapability(suite.A().Chain.GetSimApp().ScopedIBCMockKeeper, suite.A().ChannelConfig.PortID, suite.A().ChannelID)
+			chanCap = suite.A().Chain.GetChannelCapability(suite.A().ChannelConfig.PortID, suite.A().ChannelID)
+			packetHeight = suite.Z().Chain.LastHeader.GetHeight()
+			queryMultihopProofExpectedToFail = true
+			doUpdateClient = false
```

Comment:
I think this is no longer needed for this scenario

Reply:
Fixed